src/physlearn/utils.py

```diff
@@ -0,0 +1,266 @@
+    time_axis = np.arange(start, end, dt).astype(datetime)
+    return time_axis
+
+
+def find_val_in_list(my_list: Sequence, value):
+    """
+    Finds a value in a given list.
+
+    Args:
+      my_list: a Sequence to search in
+      value: to search for in the given list
+
+      Returns:
+           all indices of a given value in a given list.
+
+    """
+    return tuple(i for i, e in enumerate(my_list) if e == value)
+
+
+def find_files_with_extension(extension: str, files_path: Path):
+    """
+    Finds files recursively from the current path with the given extension.
+
+    Args:
+      extension: string of desired file extension ('edf', 'txt', etc)
+      files_path: path for txt files
+
+    Returns:
+      files: a list of file paths (str) with the extension
+
+    """
+    files = []
+    for file in Path(files_path).glob("**/*." + extension):
+        files.append(file)
+    return files
+
+
+def remove_duplicates_in_list(my_list: list):
+    """
+    Remove duplicates in a given list.
+
+    Args:
+      my_list: input list with duplicates
+
+    Returns:
+      my_list: the given list without duplicates.
+
+    """
+    if my_list is None:
+        return my_list
+    return list(set(my_list))
+
+
+def find_elements_by_condition(my_list: list, condition: Callable):
+    """
+    Find elements in a list with a given condition.
+
+    Args:
+      my_list: input list
+      condition: condition elements fill
+
+    Returns:
+      elements: list of elements in the given list that fulfill the condition
+
+    """
+    elements = [my_list[i] for i, element in enumerate(my_list) if condition(element)]
+    return elements
+
+
+def check_hdf5_file_validity(path):
+    """
+    Check that the given path points to an hdf5 file.
+
+    Raises:
+        Value error if the file is not an hdf5 file.
+    """
+    if not h5py.is_hdf5(path):
+        raise ValueError("The given path is not an hdf5 file")
+
+
+def datetime_into_float(
+    absolute_time: datetime, relative_start_time: datetime
+) -> float:
+    """
+    Converts a datetime object of some event into a float object indicating
+    the number of seconds from the time of the event to another relative start time.
+
+    Args:
+      absolute_time: datetime object indicating absolute date and time of an event.
+      relative_start_time: the time of the event should be converted to the
+    number of seconds from the given relative start time.
+
+    Returns:
+        relative_time: float: time difference between absolute time of the event
+    to the relative start time, in seconds.
+
+    """
+    relative_time = (absolute_time - relative_start_time).total_seconds()
+    return relative_time
+
+
+def find_digits_in_str(string: str) -> str:
+    """Find digits in a given string.
+
+    Args:
+      string: str, input string with the desired digits
+
+    Returns:
+      digits: str, found in the given string
+
+    """
+    digits = "".join(x for x in string if x.isdigit())
```

Comment:
I think the constant is redundant, just return

Reply:
Right